easybuild/tools/hooks.py

```diff
@@ -0,0 +1,139 @@
+# #
+# Copyright 2017-2017 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+# #
+"""
+Hook support.
+
+:author: Kenneth Hoste (Ghent University)
+"""
+import imp
+import os
+from vsc.utils import fancylogger
+
+from easybuild.tools.build_log import EasyBuildError
+
+
+_log = fancylogger.getLogger('hooks', fname=False)
+
+# this should be obtained via EasyBlock.get_steps(),
+# but we can't import from easybuild.framework.easyblock without introducing a cyclic dependency...
```

Comment:
But maybe we can push out this list, so we can import the same list on both sides?

Reply:
Yeah, I'm not too happy with this either...
We could keep this list somewhere else, I'll see if that a good option.